netx/httptransport/http3transport_test.go

```diff
@@ -20,6 +21,97 @@ func (d MockHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *qui
 	return quic.DialAddrEarly(host, tlsCfg, cfg)
 }
 
+type MockSNIHTTP3Dialer struct {
+	namech chan string
+}
+
+func (d MockSNIHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+	d.namech <- tlsCfg.ServerName
+	close(d.namech)
+	return quic.DialAddrEarly(host, tlsCfg, cfg)
+}
+
+type MockCertHTTP3Dialer struct {
+	certch chan *x509.CertPool
+}
+
+func (d MockCertHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+	d.certch <- tlsCfg.RootCAs
+	close(d.certch)
+	return quic.DialAddrEarly(host, tlsCfg, cfg)
+}
+
+func TestUnitHTTP3TransportSNI(t *testing.T) {
+	namech := make(chan string, 1)
+	sni := "sni.org"
+	txp := httptransport.NewHTTP3Transport(httptransport.Config{Dialer: selfcensor.SystemDialer{}, HTTP3Dialer: MockSNIHTTP3Dialer{namech: namech}, TLSDialer: dialer.TLSDialer{Config: &tls.Config{ServerName: sni}}})
+	req, err := http.NewRequest("GET", "https://www.google.com", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	resp, err := txp.RoundTrip(req)
+	if err == nil {
+		t.Fatal("expected error here")
+	}
+	if resp != nil {
+		t.Fatal("expected nil resp here")
+	}
+	if !strings.Contains(err.Error(), "certificate is valid for www.google.com, not "+sni) {
+		t.Fatal("unexpected error type", err)
+	}
+	servername := <-namech
+	if servername != sni {
+		t.Fatal("unexpected server name", servername)
+	}
+}
+
+func TestUnitHTTP3TransportSNINoVerify(t *testing.T) {
+	namech := make(chan string, 1)
+	sni := "sni.org"
+	txp := httptransport.NewHTTP3Transport(httptransport.Config{Dialer: selfcensor.SystemDialer{}, HTTP3Dialer: MockSNIHTTP3Dialer{namech: namech}, TLSDialer: dialer.TLSDialer{Config: &tls.Config{ServerName: sni, InsecureSkipVerify: true}}})
+	req, err := http.NewRequest("GET", "https://www.google.com", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	resp, err := txp.RoundTrip(req)
+	if err != nil {
+		t.Fatal("unexpected error")
```

Comment:
```suggestion
		t.Fatalf("unexpected error: %+v", err)
```

(or just `t.Fatal(err)`)